crates/sui-core/src/authority.rs

```diff
@@ -3164,6 +3178,9 @@ impl AuthorityState {
 
         self.committee_store.insert_new_committee(&new_committee)?;
         let mut execution_lock = self.execution_lock_for_reconfiguration().await;
+
+        cur_epoch_store.epoch_terminated().await;
```

Comment:
Could you add a comment why this needs to be here, but not other lines?

Reply:
+1 this seems subtle

Reply:
One concern is that previously we first updated global epoch store to new one `self.epoch_store.store(new_epoch_store.clone());` and then terminated all the tasks of previous epoch, and now we are doing it in different order. I don't remember at this point implication/reasoning for this choice but need to think if anything can break because of this

Reply:
> Could you add a comment why this needs to be here, but not other lines?

I don't understand what you mean by "other lines" - the reason I'm moving it is because we now want to terminate tasks before reverting extra transactions. I will add a comment, but this doesn't seem any less arbitrary than the previous location.

@andll to your comment - To me it seems intuitively much safer to terminate all per-epoch tasks before we begin any aspect of reconfiguration. For instance, prior to this change, a per-epoch task could call `state.load_epoch_store_one_call_per_task` and receive the epoch store for the new epoch instead of the one in which it was started. This seems really scary!

The only way I can think of that this could cause a problem is if it ended some task too early, and it didn't have a chance to do some necessary work. But:
- First, there shouldn't be any such task. When we arrive at this point we have executed the end-of-epoch checkpoint. As the checkpoint was certified, it should be available to all validators, so all other validators will reconfigure as well.
- Second, even if there was such a task, I don't see why it would have been guaranteed to finish in time prior to this change either.

Reply:
Ok, I'll think more if something comes up and perhaps we can rely on antithesis baking to see if any problem pop up